tooling/WindowsUpdateVsVersionInfo.c

```diff
@@ -0,0 +1,594 @@
+/**
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+*/
+// clang-format off
+// NOLINT
+/* jscpd:ignore-start */
+
+#include <stdio.h>
+#include <windows.h>
+#include <assert.h>
+#include <string.h>
+
+/**
+ * WindowsUpdateVsVersionInfo
+ *
+ * Native C app to modify the Windows RC VS_VERSION_INFO of an EXE/DLL
+ * Params:
+ *   WindowsUpdateVsVersionInfo <file.exe/dll> <key>=<value>
+ *     file.exe/dll : Program to be updated
+ *     key          : VS_VERSION_INFO String key to be updated
+ *     value        : New key value
+ *
+ * Notes: VStudio API needs to be used so that EXE/DLL object sections get
+ * updated correctly resulting in identical length and padding, when necessary.
+ */
+
+// Structures for saving existing VS_VERSION_INFO
+// Assumes:
+//   - Up to 32 String key/values
+//   - One binary Var language/codepage pair
+
+#define MAX_KEYS 32
+
+typedef struct {
+    WORD              type;
+    int               numKeys;
+    WCHAR            *key[MAX_KEYS];
+    WCHAR            *value[MAX_KEYS];
+} STRING_TABLE;
+
+typedef struct {
+    WORD              type;
+    STRING_TABLE      stringTable;
+} STRING_FILE_INFO;
+
+typedef struct {
+    WCHAR            *varKey;
+    DWORD             var;  // Allow for one binary Lang/CodePage pair
+} VAR_FILE_INFO;
+
+typedef struct {
+    WORD              type;
+    VS_FIXEDFILEINFO  vsFileInfo;
+    STRING_FILE_INFO  stringFileInfo;
+    VAR_FILE_INFO     varFileInfo;
+} VERSION_INFO;
+
+// Declarations
+short readVSVersionInfo(VERSION_INFO *info, unsigned char * vsBuf);
+short createVSVersionInfo(VERSION_INFO *info, unsigned char * vsBuf, int bufLen);
+
+// Main
```

Comment:
Same as to the Java comment, this could be split out more with named functions.

Reply:
again the Main is performing a single function. Are we saying just create a single function for it, called form main() ?

Reply:
refactored